File: LI3-with-Yoda1617-master/c_project/contribuidor.c
#include "./headers/contribuidor.h"
#include <string.h>
#include <stdio.h>
#include <stdlib.h>

struct contribuidor{
    char* username;
    int cont;
};

Contribuidor initContribuidor(char * username){
    Contribuidor new = (Contribuidor) malloc(sizeof(struct contribuidor));
    new->username = strdup(username);
    new->cont = 1;
    return new;
}

void getUsername(Contribuidor c,char* new){
    strcpy(new,c->username);
}

void setUsername(Contribuidor c,char* user){
    c->username = user;
}

int getCont(Contribuidor c){
    return c->cont;
}

void incrCont(Contribuidor c){
    c->cont++;
}


void freeCont(Contribuidor c){
    free(c);
}

File: LI3-with-Yoda1617-master/java_project/src/main/java/engine/Contribuidor.java
package engine;

public class Contribuidor{

    private String username; //Username do Contribuidor
    private int cont; //Contribuições do Contribuidor
    private long id; // Id de contribuidor

    //Construtores
    public Contribuidor(){
        username = " ";
        cont = 0;
        id = 0;
    }

    public Contribuidor(String user, int contrib, long iden){
        this.username = user;
        this.cont = contrib;
        this.id = iden;
    }

    public Contribuidor(Contribuidor c){
        username = c.getUsername();
        cont = c.getCont();
        id = c.getId();
    }

    //gets
    public String getUsername(){
        return this.username;
    }

    public int getCont(){
        return this.cont;
    }

    public long getId(){
        return this.id;
    }

    //sets
    public void setUsername(String user){
        this.username = user;
    }

    public void setCont(int contrib){
        this.cont = contrib;
    }

    public void setId(long iden){
        this.id = iden;
    }

    //Método clone
    public Contribuidor clone(){
        return new Contribuidor();
    }

    //Método equals
    public boolean equals(Object o){
        if(this==o) return true;
        if((o==null) || o.getClass() != this.getClass()) return false;
        Contribuidor c = (Contribuidor) o;
        return c.getUsername().equals(username) &&
               c.getCont() == cont              &&
               c.getId() == id;
    }

    //Método toString
    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append("Username do Contribuidor: ").append(username).append("\n");
        sb.append("Contribuições do Contribuidor: ").append(cont).append("\n");
        sb.append("Id do Contribuidor: ").append(id).append("\n");

        return sb.toString();
    }

    public void incrCont() {
        this.cont++;
    }
}
